Setup Error #255 on Windows

This is a copy of: https://github.com/carla-simulator/carla/issues/1668

which was marked closed but not resolved.

I previously built Carla 9.6 on this machine with no problems. I had removed it.

This time, when I tried to install the same way, I accidentally used x86 which led to errors.

Now when I try to install, I get this error every time:

 D:\Carla 09_06\carla>make launch
-[Setup]: [Batch params]: --boost-toolset msvc-14.1
-[Setup]: Asynchronous jobs:  8
-[Setup]: Boost toolset:      msvc-14.1
-[Setup]: Install directory:  "D:\Carla 09_06\carla\Build\"
-[Setup]: Installing zlib...
    -[install_zlib]: [Batch params]: --build-dir "D:\Carla 09_06\carla\Build\"
09_06\carla\Build\""=="" was unexpected at this time.
make: *** [setup] Error 255

I even re installed VS 2017 with no luck. I am losing my mind considering I installed no problem a month ago, and now I am getting nowhere.  I'm using x64 command prompt, and just following the instructions for source build on Windows.

--- a/PythonAPI/carla/agents/navigation/controller.py
+++ b/PythonAPI/carla/agents/navigation/controller.py
@@ -19,11 +19,8 @@ class VehiclePIDController():
     low level control a vehicle from client side
     """
 
-    MAX_BRAKE = 0.3
-    MAX_ACCEL = 0.75
-    MAX_STEER = 0.8
 
-    def __init__(self, vehicle, args_lateral, args_longitudinal):
+    def __init__(self, vehicle, args_lateral, args_longitudinal, max_throttle=0.75, max_brake=0.3, max_steering=0.8):
         """
         Constructor method.
 
@@ -39,6 +36,10 @@ class VehiclePIDController():
             K_D -- Differential term
             K_I -- Integral term
         """
+
+        self.max_brake = max_brake
+        self.max_throt = max_throttle
+        self.max_steer = max_steering
 
         self._vehicle = vehicle
         self._world = self._vehicle.get_world()
@@ -61,11 +62,11 @@ class VehiclePIDController():
         current_steering = self._lat_controller.run_step(waypoint)
         control = carla.VehicleControl()
         if acceleration >= 0.0:
-            control.throttle = min(acceleration, self.MAX_ACCEL)
+            control.throttle = min(acceleration, self.max_throt)
             control.brake = 0.0
         else:
             control.throttle = 0.0
-            control.brake = min(abs(acceleration), self.MAX_ACCEL)
+            control.brake = min(abs(acceleration), self.max_brake)
 
         # Steering regulation: changes cannot happen abruptly, can't steer too much.
 
@@ -75,9 +76,9 @@ class VehiclePIDController():
             current_steering = self.past_steering - 0.1
 
         if current_steering >= 0:
-            steering = min(self.MAX_STEER, current_steering)
+            steering = min(self.max_steer, current_steering)
         else:
-            steering = max(-self.MAX_STEER, current_steering)
+            steering = max(-self.max_steer, current_steering)
 
         control.steer = steering
         control.hand_brake = False

--- a/PythonAPI/carla/agents/navigation/local_planner.py
+++ b/PythonAPI/carla/agents/navigation/local_planner.py
@@ -81,13 +81,11 @@ class LocalPlanner(object):
     def __del__(self):
         if self._vehicle:
             self._vehicle.destroy()
-        if self._debug:
             print("Destroying ego-vehicle!")
 
     def reset_vehicle(self):
         self._vehicle = None
-        if self._debug:
-            print("Resetting ego-vehicle!")
+        print("Resetting ego-vehicle!")
 
     def _init_controller(self, opt_dict):
         """
@@ -97,7 +95,6 @@ class LocalPlanner(object):
         :return:
         """
         # default params
-        self._debug = True
         self._dt = 1.0 / 20.0
         self._target_speed = 20.0  # Km/h
         self._sampling_radius = self._target_speed * 1 / 3.6  # 1 seconds horizon
@@ -135,8 +132,6 @@ class LocalPlanner(object):
                 self._max_brake = opt_dict['max_brake']
             if 'max_steering' in opt_dict:
                 self._max_steer = opt_dict['max_steering']
-            if 'debug' in opt_dict:
-                self._debug = opt_dict['debug']
 
         self._current_waypoint = self._map.get_waypoint(self._vehicle.get_location())
         self._vehicle_controller = VehiclePIDController(self._vehicle,

--- a/PythonAPI/carla/agents/navigation/local_planner.py
+++ b/PythonAPI/carla/agents/navigation/local_planner.py
@@ -81,11 +81,13 @@ class LocalPlanner(object):
     def __del__(self):
         if self._vehicle:
             self._vehicle.destroy()
-        print("Destroying ego-vehicle!")
+        if self._debug:
+            print("Destroying ego-vehicle!")
 
     def reset_vehicle(self):
         self._vehicle = None
-        print("Resetting ego-vehicle!")
+        if self._debug:
+            print("Resetting ego-vehicle!")
 
     def _init_controller(self, opt_dict):
         """
@@ -95,10 +97,14 @@ class LocalPlanner(object):
         :return:
         """
         # default params
+        self._debug = True
         self._dt = 1.0 / 20.0
         self._target_speed = 20.0  # Km/h
         self._sampling_radius = self._target_speed * 1 / 3.6  # 1 seconds horizon
         self._min_distance = self._sampling_radius * self.MIN_DISTANCE_PERCENTAGE
+        self._max_brake = 0.3
+        self._max_throt = 0.75
+        self._max_steer = 0.8
         args_lateral_dict = {
             'K_P': 1.95,
             'K_D': 0.2,
@@ -123,11 +129,22 @@ class LocalPlanner(object):
                 args_lateral_dict = opt_dict['lateral_control_dict']
             if 'longitudinal_control_dict' in opt_dict:
                 args_longitudinal_dict = opt_dict['longitudinal_control_dict']
+            if 'max_throttle' in opt_dict:
+                self._max_throt = opt_dict['max_throttle']
+            if 'max_brake' in opt_dict:
+                self._max_brake = opt_dict['max_brake']
+            if 'max_steering' in opt_dict:
+                self._max_steer = opt_dict['max_steering']
+            if 'debug' in opt_dict:
+                self._debug = opt_dict['debug']
 
         self._current_waypoint = self._map.get_waypoint(self._vehicle.get_location())
         self._vehicle_controller = VehiclePIDController(self._vehicle,
                                                         args_lateral=args_lateral_dict,
-                                                        args_longitudinal=args_longitudinal_dict)
+                                                        args_longitudinal=args_longitudinal_dict,
+                                                        max_throttle=self._max_throt,
+                                                        max_brake=self._max_brake,
+                                                        max_steering=self._max_steer,)
 
         self._global_plan = False
 
@@ -162,7 +179,9 @@ class LocalPlanner(object):
             last_waypoint = self._waypoints_queue[-1][0]
             next_waypoints = list(last_waypoint.next(self._sampling_radius))
 
-            if len(next_waypoints) == 1:
+            if len(next_waypoints) == 0:
+                break
+            elif len(next_waypoints) == 1:
                 # only one option available ==> lanefollowing
                 next_waypoint = next_waypoints[0]
                 road_option = RoadOption.LANEFOLLOW

